test/e2e/bgppolicy_test.go

```diff
@@ -123,84 +114,177 @@ func TestBGPPolicy(t *testing.T) {
+							labelNodeHostname: clusterInfo.nodes[i].name,
+						},
+					},
+					LocalASN:   localASN,
+					ListenPort: ptr.To[int32](179),
+					Confederation: &crdv1alpha1.Confederation{
+						Identifier: int32(64512),
+					},
+					Advertisements: crdv1alpha1.Advertisements{
+						Service: &crdv1alpha1.ServiceAdvertisement{
+							IPTypes: []crdv1alpha1.ServiceIPType{crdv1alpha1.ServiceIPTypeClusterIP},
+						},
+						Pod: &crdv1alpha1.PodAdvertisement{},
+					},
+					BGPPeers: []crdv1alpha1.BGPPeer{
+						{Address: externalInfo.externalFRRIPv4, ASN: int32(65000)},
+					},
+				},
+			}
+			t.Logf("Creating a test BGPPolicy %q with ASN %d, the confederation ID 65000, applied to Node %s", bgpPolicy.Name, localASN, clusterInfo.nodes[i].name)
+			bgpPolicy, err = data.CRDClient.CrdV1alpha1().BGPPolicies().Create(context.TODO(), bgpPolicy, metav1.CreateOptions{})
+			require.NoError(t, err)
+			defer data.CRDClient.CrdV1alpha1().BGPPolicies().Delete(context.TODO(), bgpPolicyName, metav1.DeleteOptions{})
+			bgpPolicies = append(bgpPolicies, bgpPolicy)
+		}
+
+		t.Log("Getting the routes installed on the remote FRR router and verifying them")
+		expectedRoutes := make([]FRRRoute, 0)
+		for _, node := range clusterInfo.nodes {
+			expectedRoutes = append(expectedRoutes, FRRRoute{Prefix: node.podV4NetworkCIDR, Nexthops: []string{node.ipv4Addr}})
+		}
+		expectedRoutes = append(expectedRoutes, FRRRoute{Prefix: clusterIP + "/32", Nexthops: getAllNodeIPs()})
+		checkFRRRouterBGPRoutes(t, expectedRoutes, nil)
+
+		t.Log("Verifying the connectivity of the installed routes on the remote FRR route")
+		ipsToConnect := []string{podIP, clusterIP}
+		for _, ip := range ipsToConnect {
+			checkFRRRouterBGPRouteConnectivity(ip)
+		}
+
+		t.Log("Updating the BGP configuration on the remote FRR router")
+		configureExternalBGPRouter(t, int32(65000), int32(64513), false)
+
+		t.Log("Updating InternalTrafficPolicy of the test Service from Cluster to Local")
+		_, err = data.updateServiceInternalTrafficPolicy("agnhost-svc", true)
+		require.NoError(t, err)
+		defer data.updateServiceInternalTrafficPolicy("agnhost-svc", false)
+
+		for _, bgpPolicy := range bgpPolicies {
+			t.Logf("Updating the test BGPPolicy %q: setting a new confederation ID and removing Pod CIDR advertisement", bgpPolicy.Name)
+			updatedBGPPolicy := bgpPolicy.DeepCopy()
+			updatedBGPPolicy.Spec.Confederation.Identifier = int32(64513)
+			updatedBGPPolicy.Spec.Advertisements.Pod = nil
+			_, err = data.CRDClient.CrdV1alpha1().BGPPolicies().Update(context.TODO(), updatedBGPPolicy, metav1.UpdateOptions{})
+			require.NoError(t, err)
+		}
+
+		t.Log("Getting routes installed on the remote FRR router and verifying them")
+		expectedRoutes = []FRRRoute{{Prefix: clusterIP + "/32", Nexthops: []string{nodeIPv4(0)}}}
+		notExpectedRoutes := []FRRRoute{{Prefix: clusterIP + "/32", Nexthops: getAllNodeIPs()}}
+		checkFRRRouterBGPRoutes(t, expectedRoutes, notExpectedRoutes)
```

Comment:
I know this was not introduced by this PR, but I wonder why we don't just check `expectedRoutes == actualRoutes`. Are there some extra routes that we are not accounting for in `expectedRoutes` but which are still "correct"?

Reply:
When we need to advertise ClusterIPs, the Service created by Antrea will also be advertised. The ClusterIP of the Service is not included in `expectedRoutes`. That's the reason why we don't just check `expectedRoutes == actualRoutes`.